src/Components/WebAssembly/testassets/StandaloneApp/Pages/FetchDataWithCustomHttpClient.razor

```diff
@@ -0,0 +1,86 @@
+@page "/fetchdata"
+@page "/fetchdata/{StartDate:datetime}"
+@using WebAssembly.Net.Http.HttpClient
+@inject NavigationManager Nav
+
+<h1>Weather forecast</h1>
+
+<p>This component demonstrates fetching data from the server.</p>
+
+@if (forecasts == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
+    <table class='table'>
+        <thead>
+            <tr>
+                <th>Date</th>
+                <th>Temp. (C)</th>
+                <th>Temp. (F)</th>
+                <th>Summary</th>
+            </tr>
+        </thead>
+        <tbody>
+            @foreach (var forecast in forecasts)
+            {
+                <tr>
+                    <td>@forecast.DateFormatted</td>
+                    <td>@forecast.TemperatureC</td>
+                    <td>@forecast.TemperatureF</td>
+                    <td>@forecast.Summary</td>
+                </tr>
+            }
+        </tbody>
+    </table>
+    <p>
+        <a href="fetchdata/@startDate.AddDays(-5).ToString("yyyy-MM-dd")" class="btn btn-secondary float-left">
+            ◀ Previous
+        </a>
+        <a href="fetchdata/@startDate.AddDays(5).ToString("yyyy-MM-dd")" class="btn btn-secondary float-right">
+            Next ▶
+        </a>
+    </p>
+}
+
+@code {
+    [Parameter] public DateTime? StartDate { get; set; }
+
+    WeatherForecast[] forecasts;
+    DateTime startDate;
+
+    protected override async Task OnParametersSetAsync()
+    {
+        var handler = new WasmHttpMessageHandler
+        {
+            FetchRequestOptions =
+            {
+                Credentials = FetchCredentialsOption.Include,
+            }
+        };
+        var httpClient = new HttpClient(handler)
```

Comment:
This is admittedly verbose. We can author some factory code to construct this, but it needs to access the `NavigationManger`. A couple of options if we want to explore this:

a) We could add a factory that needs to be resolved from DI: `WebAssemblyHttpClientFactory`
b) We could have the user pass in the nav manager or IServiceProvider. Feels kinda weird
c) We could be sneaky and access the [NavManager.Instance](https://github.com/dotnet/aspnetcore/blob/blazor-wasm/src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyNavigationManager.cs#L18) property. This might allow writing: `var client = WebAsssemblyHttpClientFactory.Create(options: ...);` where the factory is just a static type.